web/src/beta/components/Modal/index.tsx

```diff
@@ -1,54 +1,163 @@
-import { ReactNode } from "react";
+import { ReactNode, useState } from "react";
 
 import Wrapper from "@reearth/beta/components/Modal/ModalFrame";
-import Text from "@reearth/beta/components/Text";
-import { styled, useTheme } from "@reearth/services/theme";
+import { styled } from "@reearth/services/theme";
+
+type Size = "sm" | "md" | "lg";
+
+export type SidebarTab = {
+  id: string;
+  label: string;
+  content: ReactNode;
+  tabButton1?: ReactNode;
+  tabButton2?: ReactNode;
+};
 
 type Props = {
   className?: string;
-  title?: string;
-  size?: "sm" | "md" | "lg";
+  modalTitle?: string;
+  size?: Size;
   button1?: ReactNode;
   button2?: ReactNode;
   children?: ReactNode;
   isVisible?: boolean;
   onClose?: () => void;
+  showSidebar?: boolean;
+  sidebarTabs?: SidebarTab[];
 };
 
 const Modal: React.FC<Props> = ({
   className,
-  title,
-  size,
+  modalTitle,
+  size = "md",
   button1,
   button2,
+  children,
   isVisible,
   onClose,
-  children,
+  sidebarTabs = [],
```

Comment:
This shouldn't have a default

Reply:
All the conditions are with length so I think it is ok to keep what do you think ?

Reply:
done